### PR TITLE
chore: use named return

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-5205](https://axelarnetwork.atlassian.net/browse/AXE-5205)
- Use the named return variables for efficient gas and consistency.
- Updated comments accordingly.

[AXE-5205]: https://axelarnetwork.atlassian.net/browse/AXE-5205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/gateway/AxelarAmplifierGateway.sol [hunk 1]
```diff
@@ -113,15 +113,15 @@ contract AxelarAmplifierGateway is BaseAmplifierGateway, BaseWeightedMultisig, U
...
      */
     function validateProof(bytes32 dataHash, Proof calldata proof) external view returns (bool isLatestSigners) {
-        return _validateProof(dataHash, proof);
+        isLatestSigners = _validateProof(dataHash, proof);
     }
 
     /**
      * @notice Returns the address of the gateway operator.
-     * @return The address of the operator.
+     * @return operatorAddress The address of the operator.
```
Comment (milapsheth) [inline contracts/gateway/AxelarAmplifierGateway.sol:121]: don't think the naming is very useful here. it's implied by the function name. we should use named variables only for non-obvious cases

File: contracts/gateway/BaseAmplifierGateway.sol [hunk 2]
```diff
@@ -224,15 +229,16 @@ abstract contract BaseAmplifierGateway is IBaseAmplifierGateway {
...
     function _messageHash(
         bytes32 commandId,
         string calldata sourceChain,
         string calldata sourceAddress,
         address contractAddress,
         bytes32 payloadHash
-    ) internal pure returns (bytes32) {
-        return keccak256(abi.encode(commandId, sourceChain, sourceAddress, contractAddress, payloadHash));
+    ) internal pure returns (bytes32 hash) {
+        hash = keccak256(abi.encode(commandId, sourceChain, sourceAddress, contractAddress, payloadHash));
```
Comment (milapsheth) [inline contracts/gateway/BaseAmplifierGateway.sol:241]: same, doesn't need to be named. although good to have the natspec doc

File: contracts/governance/BaseMultisig.sol [hunk 3]
```diff
@@ -53,18 +53,18 @@ contract BaseMultisig is IBaseMultisig {
 
     /**
      * @notice Returns the current signer threshold
-     * @return uint The signer threshold
+     * @return threshold The signer threshold
      */
-    function signerThreshold() external view override returns (uint256) {
-        return signers.threshold;
+    function signerThreshold() external view override returns (uint256 threshold) {
```
Comment (milapsheth) [inline contracts/governance/BaseMultisig.sol:58]: same, avoid all these obvious cases

File: contracts/gateway/AxelarAmplifierGateway.sol (hunk 1 above)
Comment (milapsheth) [reply contracts/gateway/AxelarAmplifierGateway.sol:121]: this is another example of not taking the report at face value. we should only make the change that actually makes sense to us

Comment (ahramy) [reply contracts/gateway/AxelarAmplifierGateway.sol:121]: the report says modify only one line for `axelar-gmp-sdk-solidity`, but I reviewed all files and thought it's better to be consistent with the pattern (using named return). I agree with that `we should use named variables only for non-obvious cases`. Reviewed all files and updated.

File: contracts/gateway/BaseAmplifierGateway.sol (hunk 2 above)
Comment (ahramy) [reply contracts/gateway/BaseAmplifierGateway.sol]: updated.

File: contracts/governance/BaseMultisig.sol (hunk 3 above)
Comment (ahramy) [reply contracts/governance/BaseMultisig.sol]: updated.

File: contracts/governance/InterchainGovernance.sol [hunk 4]
```diff
@@ -174,7 +174,8 @@ contract InterchainGovernance is AxelarGMPExecutable, TimeLock, Caller, IInterch
     }
 
     /**
-     * @dev Get proposal hash using the target, callData, and nativeValue
+     * @notice Get proposal hash using the target, callData, and nativeValue
+     * @return The proposal hash
```
Comment (milapsheth) [inline contracts/governance/InterchainGovernance.sol:178]: This is another contract fixed in production. Prefer to avoid changing it unless needed

**Review by milapsheth — APPROVED** [review]

File: contracts/governance/InterchainGovernance.sol (hunk 4 above)
Comment (ahramy) [reply contracts/governance/InterchainGovernance.sol:178]: Got it. Reverted change.

**Pull request merged**